src/Illuminate/Database/DatabaseTransactionsManager.php

```diff
@@ -26,6 +26,19 @@ class DatabaseTransactionsManager
     public function __construct()
     {
         $this->transactions = collect();
+        $this->afterCommitCallbacksInTestMode = false;
+    }
+
+    /**
+     * Sets the transaction manager to test mode.
+     *
+     * @return self
+     */
+    public function withAfterCommitCallbacksInTestMode()
```

Comment:
*`TestMode`* naming seems a bit imprecise. Not all tests use `RefreshDatabase` (and `DatabaseTransactions`), so wouldn't something like ***`DatabaseTransactionsTestMode`*** be more descriptive? Since this method is automatically called by the trait initialization process rather than by the user, I don't think we need to worry about the verbosity of the name.

The other perspective is LGTM!

Reply:
What name do you suggest?

Reply:
Oh, on a re-read, it seems you're suggesting renaming the class. If so, I'm not sure about that. I thought about renaming it to `DatabaseTransactionCallbacksManager` or something, but renaming the class would be a bigger change to the FW itself.

Reply:
No no, I'm just suggesting the method renaming. How about `withAfterCommitCallbacksInDatabaseTransactionsTestMode`? It's a bit verbose but very clear.

Reply:
Done